libraries/abstractions/secure_sockets/include/iot_secure_sockets.h

```diff
@@ -253,6 +253,33 @@ Socket_t SOCKETS_Socket( int32_t lDomain,
                          int32_t lProtocol );
 /* @[declare_secure_sockets_socket] */
 
+/**
+ * @brief Bind a TCP socket.
```

Comment:
Having a preconfigured source port allows customers to bind to the specified local port instead of ephemeral port for security and packet filter reasons.

We plan to run server applications on Cypress Kits so we would like to have `SOCKETS_Listen()` API implemented in future.

The limitations are that caller of `SOCKETS_Bind()` API  should make sure the socket address has the correct local IP address for the interface.

The LWIP stack has ephemeral port range from (0xc000) to (0xffff), if the socket binds to a particular port source port and before connect is called, then if the chosen port is between the ephemeral port range the API  `tcp_new_port() ` may pick the source port ( by scanning its internal TCP connection list ) the source port which is used by 
`SOCKETS_Bind() ` API before connect.

The above comments will be updated in the comments section of the `SOCKETS_Bind()` API.